src/components/Users/UserForm.tsx

```diff
@@ -160,65 +201,85 @@ export default function CreateUserForm({ onSubmitSuccess }: Props) {
     }
   };
 
+  const { mutate: createUser, isPending: createPending } = useMutation({
+    mutationKey: ["create_user"],
+    mutationFn: mutate(userApi.create),
+    onSuccess: (resp: UserBase) => {
+      toast.success(t("user_added_successfully"));
+      onSubmitSuccess?.(resp);
+    },
+    onError: (error) => {
+      toast.error(error?.message ?? t("user_add_error"));
+    },
+  });
+
+  const { mutate: updateUser, isPending: updatePending } = useMutation({
+    mutationKey: ["update_user"],
+    mutationFn: mutate(userApi.update, {
+      pathParams: { username: existingUsername! },
+    }),
+    onSuccess: (resp: UserBase) => {
+      toast.success(t("user_updated_successfully"));
+      onSubmitSuccess?.(resp);
+    },
+    onError: (error) => {
+      toast.error(error?.message ?? t("user_update_error"));
+    },
+  });
```

Comment:
we should be invalidating the user list and user retrieve queries here

Reply:
Yep, I added the user one based on #10045 (which is converting the user retrieve to using tanstack).